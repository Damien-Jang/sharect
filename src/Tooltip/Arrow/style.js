const style = ({ arrowSize, backgroundColor, buttonSize, icons }) =>
  'position:absolute;' +
  'border-left:' + arrowSize + 'px solid transparent;' +
  'border-right:' + arrowSize + 'px solid transparent;' +
  'border-top:' + arrowSize + 'px solid ' + backgroundColor + ';' +
  'bottom:-' + (arrowSize - 1) + 'px;' +
  'left:' + ((buttonSize * icons.length) / 2 - arrowSize) + 'px;' +
  'width:0;' +
  'height:0;'

export default style
