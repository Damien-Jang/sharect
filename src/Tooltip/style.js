export const defaultStyle = backgroundColor =>
  "line-height:0;" +
  "transition:all .2s ease-in-out;" +
  "background:" + backgroundColor + ";"

export const mobileStyle = buttonSize =>
  "position:fixed;" +
  "bottom:calc(50% - 64px);" +
  "left:0;" +
  "width:" + buttonSize + "px;" +
  "border-top-right-radius:5px;" +
  "border-bottom-right-radius:5px;"

export const desktopStyle = (top, left) =>
  "position:absolute;" +
  "border-radius:3px;" +
  "top:" + top + "px;" +
  "left:" + left + "px;"
