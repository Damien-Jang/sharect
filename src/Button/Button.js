import style from "./style";

function handleMouseOver() {
  this.style.transform = "scale(1.2)";
}

function handleMouseOut() {
  this.style.transform = "scale(1)";
}

export default function createButton(icon, handleMouseDown) {
  const btn = document.createElement("div");
  btn.style.cssText = style;
  btn.innerHTML = icon;
  btn.onmousedown = handleMouseDown;
  btn.onmouseover = handleMouseOver;
  btn.onmouseout = handleMouseOut;

  return btn;
}
