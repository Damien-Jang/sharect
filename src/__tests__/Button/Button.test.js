import Button from '../../Button/Button'
import style from '../../Button/style'

describe('Button', () => {
  it('should render a button element', () => {
    const button = Button('icon')
    expect(button).toMatchSnapshot()
  })

  it('should call handleMouseDown on mousedown', () => {
    const handleMouseDown = jest.fn()
    const button = Button('icon', handleMouseDown)
    button.onmousedown()

    expect(handleMouseDown).toHaveBeenCalled()
  })
  
  it('should contain an icon in innerHTML', () => {
    const button = Button('icon')

    expect(button.innerHTML).toBe('icon')
  })

  it('should have style', () => {
    const button = Button('icon')

    const styledButton = document.createElement('div')
    styledButton.style.cssText = style

    expect(button.style.cssText).toEqual(styledButton.style.cssText)
  })  

  it('should grow transform scale on mouseover', () => {
    const button = Button('icon')
    button.onmouseover()

    expect(button.style.transform).toEqual("scale(1.2)")
  })

  it('should return to the initial transform scale on mouseout', () => {
    const button = Button('icon')
    button.onmouseout()

    expect(button.style.transform).toEqual("scale(1)")
  })
  
})
