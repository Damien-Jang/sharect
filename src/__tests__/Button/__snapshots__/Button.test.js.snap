// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a button element 1`] = `
<div
  style="display: inline-block; margin: 7px; cursor: pointer; transition: all .2s ease-in-out;"
>
  icon
</div>
`;
